lua/graph.lua

```diff
@@ -1831,22 +1845,41 @@ function nodes_proc (kv, nokeys)
 
     if kv["public_transport"] == nil and kv["name"] then
        kv["junction"] = "named"
+    else
+       named_jct = false
     end
   end
 
   if kv["public_transport"] == nil and kv["name"] then
     if kv["highway"] == "traffic_signals" then
        if kv["junction"] ~= "yes" then
           kv["junction"] = "named"
+       else
+          named_jct = false
+       end
+    else
+       if kv["junction"] == "yes" or kv["reference_point"] == "yes" then
+         kv["junction"] = "named"
+       else
+          named_jct = false
        end
-    elseif kv["junction"] == "yes" or kv["reference_point"] == "yes" then
-       kv["junction"] = "named"
     end
   end
 
   --store a mask denoting access
   kv["access_mask"] = bit32.bor(auto, emergency, truck, bike, foot, wheelchair, bus, hov, moped, motorcycle, taxi)
 
+  --do not store exit_info and ref if highway ~= motorway_junction
+  if kv["highway"] ~= "motorway_junction" then
+    kv["exit_to"] = nil
+    kv["ref"] = nil
+  end
+
+  --do not store name if named_jct == false and highway ~= motorway_junction
+  if (named_jct ~= true and kv["highway"] ~= "motorway_junction") then
```

Comment:
@kdiluca Wonder if this should be a `or` here.  What do you think

Reply:
@gknisely  I believe we need to keep it `and` but we're still missing 10.
Master: 
```
2020/03/02 14:51:57.520537 [INFO] Number of nodes with refs (exits) = 1326
2020/03/02 14:51:57.520545 [INFO] Number of nodes with exit_to = 0
2020/03/02 14:51:57.520551 [INFO] Number of nodes with names = 127
2020/03/02 14:51:57.520558 [INFO] Number of way refs = 38536
2020/03/02 14:51:57.520564 [INFO] Number of reverse way refs = 10469
```
Code change `if (named_jct ~= true and kv["highway"] ~= "motorway_junction") then` : 
```
2020/02/28 20:59:21.630673 [INFO] Number of nodes with refs (exits) = 1326
2020/02/28 20:59:21.630685 [INFO] Number of nodes with exit_to = 0
2020/02/28 20:59:21.630696 [INFO] Number of nodes with names = 117
2020/02/28 20:59:21.630706 [INFO] Number of way refs = 38536
2020/02/28 20:59:21.630715 [INFO] Number of reverse way refs = 10469
```
Code change `if (named_jct ~= true or kv["highway"] ~= "motorway_junction") then`
```
2020/03/02 15:40:26.003763 [INFO] Number of nodes with refs (exits) = 1326
2020/03/02 15:40:26.003767 [INFO] Number of nodes with exit_to = 0
2020/03/02 15:40:26.003771 [INFO] Number of nodes with names = 0
2020/03/02 15:40:26.003775 [INFO] Number of way refs = 38536
2020/03/02 15:40:26.003779 [INFO] Number of reverse way refs = 10469
```

Reply:
Ok.  sounds like you will have to debug this one. @kdiluca